pkg/utils/hwaddr/hwaddr.go

```diff
@@ -51,18 +54,10 @@ func GenerateHardwareAddr4(ip net.IP, prefix string) (net.HardwareAddr, error) {
 		}
 	}
 
-	mac := prefix
-	sections := strings.Split(ip.String(), ".")
-	for _, s := range sections {
-		i, _ := strconv.Atoi(s)
-		mac += fmt.Sprintf(":%02x", i)
-	}
-
-	hwAddr, err := net.ParseMAC(mac)
-	if err != nil {
-		return nil, MacParseErr{msg: fmt.Sprintf(
-			"Failed to parse mac address %q generated based on IP %q due to: %v", mac, ip, err),
-		}
-	}
-	return hwAddr, nil
+	ipByteLen := len(ip)
```

Comment:
Well, except that the last bytes are extracted, so it doesn't matter if there are the extra null bytes. So, it works, and I was wrong.